nextcord/automod.py

```diff
@@ -0,0 +1,366 @@
+    timeout_seconds: Optional[:class:`int`]
+        The number of seconds this rule should timeout when someone triggered the rule.
+    """
+
+    def __init__(self, data: AutoModerationActionPayload):
+        self.type: ActionType = try_enum(ActionType, data["type"])
+        metadata = data.get("metadata", {})
+        self.notify_channel_id: Optional[int] = (
+            int(metadata.get("channel_id")) if metadata.get("channel_id") is not None else None
+        )
+        self.timeout_seconds: Optional[int] = metadata.get("duration_seconds")
+
+    def __repr__(self):
+        return f"<AutoModerationAction type={self.type}, notify_channel_id={self.notify_channel_id}, timeout_seconds={self.timeout_seconds}>"
+
+
+class AutoModerationRule(Hashable):
+    """Represents an auto moderation rule.
+
+    .. container:: operations
+        .. describe:: x == y
+            Check if two auto moderation rules are equal.
+        .. describe:: x != y
+            Check if two auto moderation rules are not equal.
+
+    Attributes
+    -----------
+    id: :class:`int`
+        The auto moderation rule's ID.
+    guild: :class:`Guild`
+        The guild that this auto moderation rule active in.
+    name: :class:`str`
+        The name of this auto moderation rule.
+    event_type: :class:`EventType`
+        The type of the event that the rule should execute when this event is called.
+    trigger_type: :class:`TriggerType`
+        The trigger type of this auto moderation rule.
+    guild_id: :class:`int`
+        The guild ID that this auto moderation rule is active in.
+    creator_id: :class:`int`
+        The creator of this auto moderation rule's ID.
+    enabled: :class:`bool`
+        Whether is this rule enabled or not.
+    exempt_role_ids: List[:class:`int`]
+        A list of roles that will not be affected by this rule.
+        .. note::
+            Bots are always not affected by any rule.
+    exempt_channel_ids: List[:class:`int`]
+        A list of channels that will not be affected by this rule.
+    keyword_filters: Optional[List[:class:`str`]]
+        The custom filters for this auto moderation rule. `None` if not set.
+    presets: Optional[List[:class:`KeywordPresetType`]]
+        The pre-set types of this auto moderation rule. `None` if not set.
+    actions: List[:class:`AutoModerationAction`]
+        The actions that this auto moderation rule will execute if triggered.
+    """
+
+    def __init__(self, *, state: ConnectionState, guild: Guild, data: AutoModerationRulePayload):
+        self.id: int = int(data["id"])
+        self.guild: Guild = guild
+        self._state: ConnectionState = state
+        self.guild_id: int = int(data["guild_id"])
+        self.name: str = data["name"]
+        self.creator_id: int = int(data["creator_id"])
+        self.event_type: EventType = try_enum(EventType, data["event_type"])
+        self.trigger_type: TriggerType = try_enum(TriggerType, data["trigger_type"])  # same
```

Comment:
Whats #same?

Reply:
```suggestion
        self.trigger_type: TriggerType = try_enum(TriggerType, data["trigger_type"])
```
type ignore but fixed